extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampStreamExtractor.java

```diff
@@ -57,6 +58,10 @@ public void onFetchPage(@Nonnull final Downloader downloader)
             // In this case, we are actually viewing an album page!
             throw new ExtractionException("Page is actually an album, not a track");
         }
+
+        if (albumJson.getArray("trackinfo").getObject(0).isNull("file")) {
```

Comment:
Are you sure that all tracks that have no streamable track are paid tracks? Can't the reason be different in some cases?

Reply:
The other possible reason is that it is not available yet because the album is currently up for pre-order. For example, this album is right now: https://tamarmitchell.bandcamp.com/album/samos-return

However it is impossible to see whether all of the tracks can be played for free after the album is released, or whether the tracks will require purchase to play. Therefore I think it is an okay option to throw this exception in either case.

Reply:
I guess otherwise we could test for `unreleased_track` and throw a different exception if that is true. Is there an appropriate exception for this case? Otherwise I would stick with `PaidContentException`.

Reply:
Personally, I think `ContentNotAvailableException` is more appropriate, since the track is currently unavailable until the album is released, whether it's purchased or not.

However, it seems like the tracks that can't be played due to the album being in pre-order do not have their own pages yet. Therefore, I don't think it's a case that can occur in the `StreamExtractor`. Going to the album in NewPipe and clicking on one of those tracks causes an error because the URL for them gets set to `https://tamarmitchell.bandcamp.com/null`, so maybe it should be handled differently in the `PlaylistExtractor`?

Reply:
I see. I think I noticed the concatenation with `null` before but was unsure about the best way to fix it.